### PR TITLE
Add documentation for promise and network middleware

### PR DESCRIPTION
**Pull request by peterhass**
https://github.com/goodhood-eu/nebenan-redux-tools/tree/documentation

https://favro.com/organization/aee06c3fe6503c804f507dd9/22c630cc02b09f2d6f98b8b3?card=Goo-35168

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,4 +1,414 @@
...
+  request: {
+     url: "/users",
+     type: 'query',
+     query: { q: 'Peter' } 
+  }
+}
+```
+
+
+### `delete` / `head` / `options` / `post` / `put` / `patch`
```
Comment (max-degterev) [inline README.md]: i think we only ever use `delete / post / put`, perhaps there is no need to document other methods?

File: README.md [hunk 2]
```diff
@@ -1,4 +1,414 @@
...
+  // item ids for current page  
+  collection: [],
+}
+```
+
+### Actions
+
+Pagination query params will be added for `query` requests with either `request.pagination.first` or `request.pagination.last` set.
+
+*Only works for endpoints supporting `lower`/`higher` params.*
```
Comment (max-degterev) [inline README.md:285]: do we need to document how to increase page parameter in the default pagination?

**Review by max-degterev — APPROVED** [review]
Good work, merge if you like

File: README.md (hunk 2 above)
Comment (peterhass) [reply README.md:285]: The page parameter is updated by `paginationGenerator`. No need to do anything

**Pull request merged**